up_client_socket/python/socket_transport.py

```diff
@@ -200,17 +196,22 @@ def invoke_method(self, method_uri: UUri, request_payload: UPayload, options: Ca
         """
         Invokes a method with the provided URI, request payload, and options.
         """
-        attributes = UAttributesBuilder.request(RESPONSE_URI, method_uri, UPriority.UPRIORITY_CS4, options.ttl).build()
+        umsg = UMessageBuilder.request(RESPONSE_URI, method_uri, options.timeout).build_from_upayload(request_payload)
         # Get uAttributes's request id
-        request_id = attributes.id
+        request_id = umsg.attributes.id
 
         response = Future()
         self.reqid_to_future[request_id.SerializeToString()] = response
         # Start a thread to count the timeout
         timeout_thread = threading.Thread(target=timeout_counter, args=(response, request_id, options.ttl))
         timeout_thread.start()
 
-        umsg = UMessage(payload=request_payload, attributes=attributes)
         self.send(umsg)
 
         return response
+
+    def get_source(self) -> UUri:
+        """
+        Returns the source URI of the UTransport.
+        """
+        return RESPONSE_URI
```

Comment:
Here instead of a fixed RESPONSE_URI for get_source method, I was suggested that transport should be able to take a input parameter of source and let end users enter their source Uri information. and this method should return that source Uri. @neelam-kushwah  do you agree with this that source information must come from end user?

Reply:
fixed, added source as an init for java and python